internal/services/mssqlmanagedinstance/mssql_managed_instance_resource.go

```diff
@@ -6,11 +6,13 @@ package mssqlmanagedinstance
 import (
 	"context"
 	"fmt"
+	"log"
 	"regexp"
 	"strings"
 	"time"
 
 	"github.com/Azure/azure-sdk-for-go/services/preview/sql/mgmt/v5.0/sql" // nolint: staticcheck
+	"github.com/gofrs/uuid"
```

Comment:
I assume that we could not use HashiCorp's UUID here instead of `github.com/gofrs/uuid`. This if because `Sid` is  of type `*uuid.UUID`, and`uuid.FromString` returns `uuid.UUID` which I didn't find a similar function to replace it in HashiCorp's UUID library. 
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/39109137/b7e50f8f-d1bf-4a38-8489-9f3e4701da96)

Reply:
Ah ok, no problem let's stick with the same library as the SDK - with a view to changing when this resource is migrated to `hashicorp/go-azure-sdk`.